bundle/edu.gemini.qpt.client/src/main/java/edu/gemini/qpt/core/listeners/LimitsListener.java

```diff
@@ -27,6 +35,19 @@
 	@SuppressWarnings("unused")
 	private static final Logger LOGGER = Logger.getLogger(LimitsListener.class.getName());
 
+	private static final DateTimeFormatter dateFormat(TimeZone zone) {
+		return DateTimeFormatter.ofPattern("HH:mm").withZone(zone.toZoneId());
+	};
+
+	private static final String formatInterval(TimeZone zone, Interval interval) {
+
+		return String.format(
+			"(%s, %s)",
+			DateTimeFormatter.ofPattern("HH:mm").withZone(zone.toZoneId()).format(Instant.ofEpochMilli(interval.getStart())),
+			DateTimeFormatter.ofPattern("HH:mm").withZone(zone.toZoneId()).format(Instant.ofEpochMilli(interval.getEnd()))
```

Comment:
Did you mean `dateFormat(zone).format(Instant.ofEpochMilli(interval.getStart()))` etc?

Reply:
I think lines 38-40 should be removed entirely, and instead, make `formatInterval` as follows:

````
private static final String formatInterval(TimeZone zone, Interval interval) {
    final Function<Long, String> f =
            ms -> DateTimeFormatter.ofPattern("HH:mm").withZone(zone.toZoneId()).format(Instant.ofEpochMilli(ms));
    return String.format("(%s, %s)", f.apply(interval.getStart()), f.apply(interval.getEnd()));
}
````

That way, you avoid instantiating and initializing the `DateTimeFormatter` twice, get rid of the otherwise useless function, and instead use a lambda.